File: .claude/skills/js-cli-beautifier/scripts/complete-example.js
#!/usr/bin/env node

/**
 * Complete CLI Beautification Example
 *
 * This script demonstrates proper implementation of all CLI beautification features
 * with correct imports and error handling to avoid common issues.
 */

// Proper imports to avoid the issues encountered
const chalk = require('chalk');
const Table = require('cli-table3');
const cliProgress = require('cli-progress');
const boxen = require('boxen');  // Note: not destructured as {boxen}
const ora = require('ora');

// Enhanced header with styling
console.log(chalk.bold.blue('╔' + '═'.repeat(60) + '╗'));
console.log(chalk.bold.blue('║') + chalk.bold.bgBlue.white(' CLI BEAUTIFIER COMPLETE EXAMPLE ') + chalk.bold.blue('║'));
console.log(chalk.bold.blue('╚' + '═'.repeat(60) + '╝'));

// Enhanced message types with color coding
console.log('');
console.log(chalk.bold('ENHANCED MESSAGE TYPES:'));
console.log(chalk.green('✓ Success: Operation completed successfully'));
console.log(chalk.red('✗ Error: An error occurred during processing'));
console.log(chalk.yellow('⚠ Warning: This is a warning message'));
console.log(chalk.blue('ℹ Info: Here is some important information'));
console.log(chalk.gray('⚙ Debug: Debug information for troubleshooting'));

// Enhanced table using cli-table3
console.log('');
console.log(chalk.bold('ENHANCED TABLE:'));

const table = new Table({
    head: [chalk.bold.blue('Name'), chalk.bold.blue('Status'), chalk.bold.blue('Last Updated')],
    colWidths: [15, 15, 15],
    style: {
        head: ['blue', 'bold'],
        border: ['grey']
    }
});

table.push(
    [chalk.green('Project A'), chalk.green.bold('Active'), '2026-01-06'],
    [chalk.yellow('Project B'), chalk.yellow.bold('Warn'), '2026-01-05'],
    [chalk.red('Project C'), chalk.red.bold('Error'), '2026-01-04']
);

console.log(table.toString());

// Enhanced progress indicator using cli-progress
console.log('');
console.log(chalk.bold('ENHANCED PROGRESS INDICATION:'));

// Create a new progress bar
const progressBar = new cliProgress.SingleBar({
    format: 'Processing items |' + chalk.cyan('{bar}') + '| {percentage}% | {value}/{total} | {msg}',
    barCompleteChar: '\u25A0',
    barIncompleteChar: '\u25A1',
    hideCursor: true
});

// Initialize the bar - (total, starting value)
progressBar.start(5, 0);

// Simulate processing items with the progress bar
for (let i = 1; i <= 5; i++) {
    // Update the bar value
    progressBar.update(i, {
        msg: `Processing item ${i}/5`
    });

    // Simulate processing time
    const startTime = Date.now();
    while (Date.now() - startTime < 300); // 300ms delay
}

// Stop the bar
progressBar.stop();
console.log(chalk.green('\n✓ All items processed successfully!'));

// Example of using spinner for async operations
console.log('');
console.log(chalk.bold('SPINNER EXAMPLE:'));

// In a real async context, you would use:
// const spinner = ora('Loading...').start();
// try {
//   await asyncOperation();
//   spinner.succeed('Operation completed!');
// } catch (error) {
//   spinner.fail('Operation failed!');
// }

// For demonstration purposes in this script:
console.log(chalk.yellow('→ Simulating async operation with spinner...'));
console.log(chalk.green('✓ Async operation completed!'));

// Enhanced status display with styled formatting
console.log('');
console.log(chalk.bold('ENHANCED STATUS INFORMATION:'));

// Create a simple bordered box manually to avoid boxen import issues
const statusContent =
    '  ' + chalk.bold.blue('Application: ') + chalk.white('CLI Beautifier Example') + '\n' +
    '  ' + chalk.bold.green('Status: ') + chalk.green('✓ Running') + '\n' +
    '  ' + chalk.bold.yellow('Version: ') + chalk.yellow('1.0.0') + '\n' +
    '  ' + chalk.bold.cyan('Uptime: ') + chalk.cyan('00:00:05') + '\n' +
    '  ' + chalk.bold.magenta('Tasks: ') + chalk.magenta('5/5 completed');

const lines = statusContent.split('\n').filter(line => line.trim() !== ''); // Remove empty lines
const maxWidth = Math.max(...lines.map(line => line.replace(/\x1B\[[0-9;]*m/g, '').length)) + 4; // Add padding and account for ANSI codes

console.log(chalk.gray('  ╔' + '═'.repeat(maxWidth - 2) + '╗'));
lines.forEach(line => {
    if (line.trim() !== '') { // Only process non-empty lines
        const cleanLineLength = line.replace(/\x1B\[[0-9;]*m/g, '').length;
        const padding = Math.max(0, maxWidth - cleanLineLength - 2); // Ensure padding is not negative
        const paddedLine = line + ' '.repeat(padding);
        console.log(chalk.gray('  ║') + paddedLine + chalk.gray('║'));
    }
});
console.log(chalk.gray('  ╚' + '═'.repeat(maxWidth - 2) + '╝'));

// Enhanced task list with better visual hierarchy
console.log('');
console.log(chalk.bold('ENHANCED TASK LIST:'));

const tasks = [
    { number: 1, title: 'Initialize CLI beautifier skill', status: '✓', color: 'green' },
    { number: 2, title: 'Test basic functionality', status: '✓', color: 'green' },
    { number: 3, title: 'Verify table output', status: '✓', color: 'green' },
    { number: 4, title: 'Check progress indicators', status: '✓', color: 'green' },
    { number: 5, title: 'Validate status display', status: '✓', color: 'green' }
];

tasks.forEach(task => {
    const colorFunction = {
        green: chalk.green,
        yellow: chalk.yellow,
        red: chalk.red
    }[task.color] || chalk.white;

    console.log(
        chalk.bold(`  ${task.number}. `) +
        colorFunction(`${task.status} ${task.title}`)
    );
});

console.log('');
console.log(chalk.bold.green('✓ All tasks completed successfully!'));

// Enhanced content container using manual border
console.log('');
console.log(chalk.bold('ENHANCED CONTENT CONTAINER:'));

const contentText = [
    chalk.bold.white('This is content in an enhanced container'),
    chalk.cyan('Multiple lines work too'),
    chalk.green('Enhanced CLI functionality is working'),
    chalk.yellow('Visual improvements applied successfully')
];

const contentWidth = Math.max(...contentText.map(line => line.replace(/\x1B\[[0-9;]*m/g, '').length)) + 4; // Account for padding and remove ANSI codes for length calculation

console.log(chalk.magenta('╔' + '═'.repeat(contentWidth) + '╗'));
contentText.forEach(line => {
    const cleanLineLength = line.replace(/\x1B\[[0-9;]*m/g, '').length;
    const padding = contentWidth - cleanLineLength - 2;
    const leftPad = Math.floor(padding / 2);
    const rightPad = padding - leftPad;
    console.log(chalk.magenta('║') + ' '.repeat(leftPad) + line + ' '.repeat(rightPad) + chalk.magenta('║'));
});
console.log(chalk.magenta('╚' + '═'.repeat(contentWidth) + '╝'));

// Enhanced completion message
console.log('');

const completionText = [
    chalk.bold.green('CLI BEAUTIFIER COMPLETE EXAMPLE!'),
    chalk.bold.white('All functionality demonstrated and ready for use.')
];

const completionWidth = Math.max(...completionText.map(line => line.replace(/\x1B\[[0-9;]*m/g, '').length)) + 4;

console.log(chalk.green('╔' + '═'.repeat(completionWidth) + '╗'));
completionText.forEach(line => {
    const cleanLineLength = line.replace(/\x1B\[[0-9;]*m/g, '').length;
    const padding = completionWidth - cleanLineLength - 2;
    const leftPad = Math.floor(padding / 2);
    const rightPad = padding - leftPad;
    console.log(chalk.green('║') + ' '.repeat(leftPad) + line + ' '.repeat(rightPad) + chalk.green('║'));
});
console.log(chalk.green('╚' + '═'.repeat(completionWidth) + '╝'));

console.log('');
console.log(chalk.bold('TROUBLESHOOTING TIPS:'));
console.log('• If boxen fails to import, use manual border drawing as shown above');
console.log('• If ora fails to import, use simple status messages as alternatives');
console.log('• Always install all dependencies: npm install chalk cli-table3 cli-progress boxen ora');

File: .claude/skills/py-cli-beautifier/scripts/complete_example.py
#!/usr/bin/env python3
"""
complete_example.py - Complete example demonstrating all Python CLI beautification features
"""

import time

try:
    from rich.console import Console
    from rich.table import Table
    from rich.progress import Progress, BarColumn, TextColumn
    from rich.panel import Panel
    from rich import print as rich_print
    RICH_AVAILABLE = True
except ImportError:
    RICH_AVAILABLE = False

try:
    from colorama import init, Fore, Back, Style
    init()  # Initialize colorama
    COLORAMA_AVAILABLE = True
except ImportError:
    COLORAMA_AVAILABLE = False

try:
    from tabulate import tabulate
    TABULATE_AVAILABLE = True
except ImportError:
    TABULATE_AVAILABLE = False

try:
    from tqdm import tqdm
    TQDM_AVAILABLE = True
except ImportError:
    TQDM_AVAILABLE = False


def print_header():
    """Print an enhanced header"""
    if RICH_AVAILABLE:
        console = Console()
        console.print(Panel("[bold blue]PYTHON CLI BEAUTIFIER COMPLETE EXAMPLE[/bold blue]", expand=False))
    else:
        print("=" * 60)
        print("PYTHON CLI BEAUTIFIER COMPLETE EXAMPLE")
        print("=" * 60)


def print_enhanced_messages():
    """Print enhanced messages with color coding"""
    print("\nENHANCED MESSAGE TYPES:")

    if RICH_AVAILABLE:
        rich_print("[green]✓ Success: Operation completed successfully[/green]")
        rich_print("[red]✗ Error: An error occurred during processing[/red]")
        rich_print("[yellow]⚠ Warning: This is a warning message[/yellow]")
        rich_print("[blue]ℹ Info: Here is some important information[/blue]")
        rich_print("[grey62]⚙ Debug: Debug information for troubleshooting[/grey62]")
    elif COLORAMA_AVAILABLE:
        print(f"{Fore.GREEN}✓ Success: Operation completed successfully{Style.RESET_ALL}")
        print(f"{Fore.RED}✗ Error: An error occurred during processing{Style.RESET_ALL}")
        print(f"{Fore.YELLOW}⚠ Warning: This is a warning message{Style.RESET_ALL}")
        print(f"{Fore.BLUE}ℹ Info: Here is some important information{Style.RESET_ALL}")
        print(f"{Fore.LIGHTBLACK_EX}⚙ Debug: Debug information for troubleshooting{Style.RESET_ALL}")
    else:
        print("✓ Success: Operation completed successfully")
        print("✗ Error: An error occurred during processing")
        print("⚠ Warning: This is a warning message")
        print("ℹ Info: Here is some important information")
        print("⚙ Debug: Debug information for troubleshooting")


def print_enhanced_table():
    """Print an enhanced table"""
    print("\nENHANCED TABLE:")

    headers = ["Name", "Status", "Last Updated"]
    rows = [
        ["Project A", "Active", "2026-01-06"],
        ["Project B", "Warn", "2026-01-05"],
        ["Project C", "Error", "2026-01-04"]
    ]

    if RICH_AVAILABLE:
        table = Table(show_header=True, header_style="bold blue")

        for header in headers:
            table.add_column(header, style="dim")

        for row in rows:
            status_color = "green" if row[1] == "Active" else "yellow" if row[1] == "Warn" else "red"
            table.add_row(row[0], f"[{status_color}]{row[1]}[/{status_color}]", row[2])

        console = Console()
        console.print(table)
    elif TABULATE_AVAILABLE:
        all_rows = [headers] + rows
        print(tabulate(all_rows, headers="firstrow", tablefmt="grid"))
    else:
        # Simple table without external libraries
        col_widths = [max(len(str(row[i])) for row in [headers] + rows) for i in range(len(headers))]
        col_widths = [max(w, len(headers[i])) for i, w in enumerate(col_widths)]

        separator = "+" + "+".join("-" * (width + 2) for width in col_widths) + "+"
        header_row = "|"
        for i, header in enumerate(headers):
            header_row += f" {header.ljust(col_widths[i])} |"

        print(separator)
        print(header_row)
        print(separator)

        for row in rows:
            data_row = "|"
            for i, cell in enumerate(row):
                data_row += f" {str(cell).ljust(col_widths[i])} |"
            print(data_row)

        print(separator)


def print_enhanced_progress():
    """Print enhanced progress indicator"""
    print("\nENHANCED PROGRESS INDICATION:")

    if TQDM_AVAILABLE:
        with tqdm(total=5, desc="Processing items", unit="item") as pbar:
            for i in range(5):
                time.sleep(0.2)  # Simulate work
                pbar.set_postfix({"item": f"{i+1}/5"})
                pbar.update(1)
    elif RICH_AVAILABLE:
        with Progress(
            TextColumn("[progress.description]{task.description}"),
            BarColumn(),
            TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
        ) as progress:
            task = progress.add_task("Processing items", total=5)
            for i in range(5):
                time.sleep(0.2)  # Simulate work
                progress.update(task, advance=1)
    else:
        # Simple progress without external libraries
        for i in range(1, 6):
            print(f"Item {i}/5 completed")
            time.sleep(0.2)
        print("Progress: 100% complete")


def print_enhanced_status():
    """Print enhanced status information"""
    print("\nENHANCED STATUS INFORMATION:")

    if RICH_AVAILABLE:
        status_content = (
            "Application: CLI Beautifier Example\n"
            "Status: [green]✓ Running[/green]\n"
            "Version: 1.0.0\n"
            "Uptime: 00:00:05\n"
            "Tasks: [magenta]5/5 completed[/magenta]"
        )
        panel = Panel(status_content, title="Status Information")
        console = Console()
        console.print(panel)
    else:
        print("  Application: CLI Beautifier Example")
        print("  Status: ✓ Running")
        print("  Version: 1.0.0")
        print("  Uptime: 00:00:05")
        print("  Tasks: 5/5 completed")


def print_enhanced_task_list():
    """Print enhanced task list"""
    print("\nENHANCED TASK LIST:")

    tasks = [
        {"number": 1, "title": "Initialize CLI beautifier skill", "status": "✓", "color": "green"},
        {"number": 2, "title": "Test basic functionality", "status": "✓", "color": "green"},
        {"number": 3, "title": "Verify table output", "status": "✓", "color": "green"},
        {"number": 4, "title": "Check progress indicators", "status": "✓", "color": "green"},
        {"number": 5, "title": "Validate status display", "status": "✓", "color": "green"},
    ]

    if RICH_AVAILABLE:
        for task in tasks:
            color_tag = task["color"]
            rich_print(f"  [bold]{task['number']}.[/bold] [{color_tag}]{task['status']} {task['title']}[/{color_tag}]")
    elif COLORAMA_AVAILABLE:
        for task in tasks:
            color_map = {
                "green": Fore.GREEN,
                "yellow": Fore.YELLOW,
                "red": Fore.RED
            }
            color = color_map.get(task["color"], Fore.WHITE)
            print(f"  \033[1m{task['number']}.\033[0m {color}{task['status']} {task['title']}{Style.RESET_ALL}")
    else:
        for task in tasks:
            print(f"  {task['number']}. {task['status']} {task['title']}")


def print_enhanced_container():
    """Print enhanced content container"""
    print("\nENHANCED CONTENT CONTAINER:")

    content = [
        "This is content in an enhanced container",
        "Multiple lines work too",
        "Enhanced CLI functionality is working",
        "Visual improvements applied successfully"
    ]

    if RICH_AVAILABLE:
        content_text = "\n".join(content)
        panel = Panel(content_text, title="Content Container")
        console = Console()
        console.print(panel)
    else:
        max_len = max(len(line) for line in content)
        border = "╔" + "═" * (max_len + 2) + "╗"
        print(border)
        for line in content:
            padding = max_len - len(line)
            print(f"║ {line}{' ' * padding} ║")
        print("╚" + "═" * (max_len + 2) + "╝")


def main():
    """Main function to demonstrate all features"""
    print_header()
    print_enhanced_messages()
    print_enhanced_table()
    print_enhanced_progress()
    print_enhanced_status()
    print_enhanced_task_list()
    print_enhanced_container()

    print("\n✓ All tasks completed successfully!")

    if RICH_AVAILABLE:
        console = Console()
        console.print(Panel("[bold green]PYTHON CLI BEAUTIFIER COMPLETE EXAMPLE![/bold green]\n[bold white]All functionality demonstrated and ready for use.[/bold white]", expand=False))
    else:
        print("\nPYTHON CLI BEAUTIFIER COMPLETE EXAMPLE!")
        print("All functionality demonstrated and ready for use.")


if __name__ == "__main__":
    main()